derive-macros/src/lib.rs

```diff
@@ -0,0 +1,93 @@
+use proc_macro2::{Ident, TokenStream};
+use quote::{quote, quote_spanned};
+use syn::spanned::Spanned;
+use syn::{parse_macro_input, Data, DataStruct, DeriveInput, Fields, PathArguments, Type};
+
+/// Derive a `deltakernel::schemas::GetField` implementation for the annotated struct. The actual
+/// field names in the schema (and therefore of the struct members) are all mandated by Delta spec,
+/// and so the user of this macro is responsible for ensuring that e.g. `Metadata::schema_string is
+/// the snake_case-ified version of `schemaString` from Delta's Change Metadata action (this macro
+/// allows the use of standard rust snake_case, and will convert to the correct delta schema
+/// camelCase version).
+#[proc_macro_derive(Schema)]
+pub fn derive_schema(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
```

Comment:
comment above added to address https://github.com/delta-incubator/delta-kernel-rs/pull/129/files#r1527171512

This will show up in the docs for the derive-macros crate as:
![image](https://github.com/delta-incubator/delta-kernel-rs/assets/178912/cd1bb93a-ba12-4ad3-88fd-aa53f5d62568)

Reply:
If you wanted you could also omit the crate from docs with `#[doc(hidden)]`

Reply:
I sort of consider the crate docs like "implementer" docs. Like someone writing a connector would read them. I think understanding what this macro does is useful from that perspective, so likely it should be left in.